### PR TITLE
AsyncElement: add trivial tests

### PR DESCRIPTION
**Pull request by eeverett6**
Add tests to AsyncElement for two trivial cases, an empty stream that
immediately returns Ready(None), and an AsyncElementLink created with an
empty channel that should panic.

This commit makes progress on #24, but does not complete it.

### CODE REVIEW

File: src/api/async_element.rs [hunk 1]
```diff
@@ -352,6 +352,44 @@ mod tests {
...
+            Ok(())
+        }));
+
+        let router_output: Vec<_> = r.iter().collect();
+        assert_eq!(router_output, packets);
+    }
+
+    #[test]
+    #[should_panic]
+    fn one_async_element_with_empty_channel() {
```
Comment (ddowl) [inline src/api/async_element.rs]: Not sure if this test is complete yet. What is the empty channel here we're testing?

File: src/api/async_element.rs [hunk 2]
```diff
@@ -352,6 +352,44 @@ mod tests {
         assert_eq!(router_output.len(), 2001);
     }
 
+    #[test]
+    fn one_async_element_with_empty_stream() {
```
Comment (ddowl) [inline src/api/async_element.rs]: This is the right idea. We have an issue open to abstract away some of the tokio running and assertion code away, but you'll necessarily have to define the input stream for your router, configure it via Element links, and assert that some stuff happened to the output stream.

File: src/api/async_element.rs (hunk 1 above)
Comment (CollinValley) [reply src/api/async_element.rs]: we are testing that if the `default_channel_size` is set to 0, it panics.

File: src/api/async_element.rs (hunk 2 above)
Comment (CollinValley) [reply src/api/async_element.rs]: This looks good to me, does what we expect.

**Review by CollinValley — DISMISSED** [review]
I think this can be merged as is. Everything looks good to me. Just resubmit the PR and we are off to the races.

**Comment by CollinValley** [comment]
Or if you want to layer on more changes to fully resolve #24 then just add them on top, your first commit looks good.

**Review by ddowl — APPROVED** [review]
